roles/keycloak/tasks/main.yml

```diff
@@ -0,0 +1,147 @@
+---
+
+- name: Include OS-specific variables
+  include_vars: "{{ item }}"
+  with_first_found:
+    - "os/{{ ansible_distribution }}-{{ ansible_distribution_release }}.yml"
+    - "os/{{ ansible_distribution }}-{{ ansible_distribution_major_version }}.yml"
+    - "os/{{ ansible_distribution }}.yml"
+    - "os/{{ ansible_os_family }}.yml"
+  tags:
+    - always
+
+- name: Include keycloak-specific variables
+  include_vars: "keycloak/{{ keycloak_version }}.yml"
+  tags:
+    - always
+
+
+### Install Java with OS-specific packages
+- include: "java/{{ ansible_distribution }}-{{ ansible_distribution_major_version }}.yml"
+  tags:
+    - always
+
+- name: Capture logged in user name
+  set_fact:
+    loggedin_user: "{{ansible_user_id}}"
+
+
+#### Extract keycloak from archive
+
+- name: Create keycloak directory
+  file:
+    path: "/tmp/keycloak-tmp"
+    state: directory
+  tags: "download_extract"
+
+- name: Download keycloak archive file
+  get_url:
+    url: "{{keycloak_archive_url}}"
+    dest: "/tmp/keycloak_archive"
+  tags: "download_extract"
+
+- name: unpack keycloak
+  unarchive:
+    src: "/tmp/keycloak_archive"
+    dest: "/tmp/keycloak-tmp"
+    copy: no
+  tags: "download_extract"
+
+- name: Find keycloak extracted folder name
+  find:
+    paths: /tmp/keycloak-tmp
+    patterns: 'keycloak*'
+    file_type: directory
+    recurse: no
+  register: find_result
+  tags: "download_extract"
+
+- name: Move extracted keycloak to final keycloak folder
+  command: "mv {{find_result.files[0].path}} {{keycloak_home}}"
+  tags: "download_extract"
+
+######## Create postgresql plugin
+
+- name: create postresql plugin dir with parent directories
+  file:
+    path: "{{keycloak_home}}/modules/system/layers/keycloak/org/postgresql/main"
+    state: directory
+  tags: "postgresql_plugin"
+
+- name: Download postgresql JDBC driver
+  get_url:
+    url: "https://jdbc.postgresql.org/download/{{postgresql_jar_name}}"
+    dest: "{{keycloak_home}}/modules/system/layers/keycloak/org/postgresql/main"
+  tags: "postgresql_plugin"
+    
+- name: Installing postgresql plugin 
+  template:
+    src: templates/module.xml.j2
+    dest: "{{keycloak_home}}/modules/system/layers/keycloak/org/postgresql/main/module.xml"
+  tags: "postgresql_plugin"
```

Comment:
Can this be moved to keycloak_plugins?

Reply:
That whould be nice, but
I'd say that this plugin is considered "systemic". I am afraid that with the current design, this cannot be moved to the plugins.
Currently, the steps are:
- Keycloak is installed (put all files in place)
- postgresql plugin is injected
- keycloak starts and admin user is created (if we remove postgres plugin, the user will be created in the embedded H2 database) 
- keycloak stops
- additional plugins are installed
- keycloak starts
- keycloak is configured

Keycloak admin user is part of the installation of keycloak (and not the configuration part, since it should be done when keycloak is not running), thus moving it to the 

I could try to create an extra yml step and move all the initialization stuff after the keycloak plugins' installation, but again, that would introduce a very compicated plugins installation play, because postgresql plugin has a completely different installation set of changes compared to the ordinary plugins (like the other 2 we already have).

I am afraid that by adding this 'special' plugin, the set of tasks in the keycloak_plugins manipulation will grow huge! 
However, i'll try to move it and see what it takes to make such a change.